test/specs/KeyboardSupportSpec.js

```diff
@@ -489,4 +489,67 @@ describe("Keyboard Support Tests", function() {
       });
     });
   });
+});
+
+describe("Navigating slider with the keyboard", function() {
+  var mySlider;
+  var keyboardEvent;
+  var options;
+  // var $handle1;
+
+  describe("Does not trigger 'change' event when values do not change", function() {
+
+    beforeEach(function() {
+      options = {
+        id: 'mySlider',
+        min: -100,
+        max: 100,
+        step: 1,
+        value: [0, 1],
+        range: true
+      };
+
+      // Create keyboard event
+      keyboardEvent = document.createEvent('Event');
+      keyboardEvent.initEvent('keydown', true, true);
+    });
+
+    afterEach(function() {
+      if (mySlider) {
+        if (mySlider instanceof Slider) { mySlider.destroy(); }
+        mySlider = null;
+      }
+    });
+
+    it("Should not trigger 'change' event", function() {
+      var hasSlideStarted = false;
+      var hasChanged = false;
+      var hasSlideStopped = false;
+      options.value = [-100, 0];
+      mySlider = new Slider($('#testSlider1')[0], options);
+      // $handle1 = $('#mySlider').find('.slider-handle:first');
+
+      mySlider.on('slideStart', function() {
+        hasSlideStarted = true;
+      });
+
+      mySlider.on('change', function() {
+        hasChanged = true;
+      });
+
+      mySlider.on('slideStop', function() {
+        hasSlideStopped = true;
+      });
+
+      // Move the handle to the left
+      keyboardEvent.keyCode = keyboardEvent.which = 37;
+
+      // FIXME: Setup async event dispatch
+      // $handle1[0].dispatchEvent(keyboardEvent);
+      mySlider.handle1Keydown(keyboardEvent);
+      expect(hasSlideStarted).toBe(true);
```

Comment:
so this may not work the way you expect it to since the callbacks within the event listeners may not be invoked until the next tick of the event loop.

I would write this test in an async manner using the `done` callback and just do the assertions within the `slideStop` event listener and invoke `done()` on completion